027-network-policy-generation-feature-gate.md

```diff
@@ -0,0 +1,63 @@
+# Network Policy Generation Feature Gate
+
+The feature would consist of a new Feature Gate which can be used to disable the generation of Network Policies by Strimzi, where the user wishes to set up a more restrictive Network Policy set.
+
+## Current situation
+
+Currently, Strimzi generates Network Policies which allow all pods in the deployment to connect to Kafka, Zookeeper and any other service that it has generated.
+
+There is no way to stop this generation.
+
+It is possible, in the select case of listeners, to further restrict the Ingress using NetworkPolicyPeers, however this does not extend to all Network Policy generation, for example Metrics.
+It is also not possible to set up Egress Network Policies. 
+
+## Motivation
+
+The current Network Policies that are generated are great for getting Strimzi up and running in a deployment where a deny-all network policy is in place.
+
+However, this does not allow users of Strimzi to reach a fully secured Network Policy set, due to Strimzi not providing the full feature set of Kubernetes Network Policies.
+
+By adding a feature gate, it allows users to disable this generation, and instead write fully secure Network Policies.
```

Comment:
I guess you should make it clear here that when the user disables the network policies generated by Strimzi, he/she is responsible for setting up the right network policies required by Strimzi to work. I think we will also need to document what these are in the implementation PR (but this proposal can mention that this will be documented).

Reply:
I agree, I will change the phrasing to be more along the lines of it becoming their responsibility, rather than they get the ability.